__tests__/app.test.js

```diff
@@ -129,6 +131,129 @@ describe('CORE: GET /api/articles/:article_id', () => {
+
+            })
+    })
+    test('returns an article array when each article has the correct keys and value types apart from comment_count', () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then((response) => {
+            const articles = response.body.articles;
+            expect(articles.length).not.toBe(0);
+
+            const exampleObj = {
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: expect.any(Number),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String)
+            }
+            articles.forEach((article) => {
+                expect(article).toMatchObject(exampleObj)
+            });
+
+        })
+    })
+    test('returns an article array when each article has all the correct keys and value types', () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then((response) => {
+            const articles = response.body.articles;
+            expect(articles.length).not.toBe(0);
+
+            const exampleObj = {
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: expect.any(Number),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(Number)
+            }
+            articles.forEach((article) => {
+                expect(article).toMatchObject(exampleObj)
+            });
+        })
+    })
+    test('In addition: the articles should be sorted by date in descending order. there should not be a body property present on any of the objects', () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then((response) => {
+            const articles = response.body.articles;
+            const orderedArticles = [...articles];
+            expect(articles.length).not.toBe(0);
+
+            orderedArticles.sort((currArt, nextArt) => Date.parse(currArt.created_at) - Date.parse(nextArt.created_at))
+            expect(orderedArticles).toEqual(articles);
+        })
+    })
+    test('returns articles without a body property present on any of the article objects' , () => {
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then((response) => {
+            const articles = response.body.articles
+            expect(articles.length).not.toBe(0);
+
+            articles.forEach((article) => {
+                expect(article).not.toHaveProperty('body');
+            })
+        })
+    })
+    test('returns a status of 404 and a custom error object of {status:404, msg:"No Articles Found"} if the database has no entries in articles table', ()=> {
```

Comment:
if there are no articles, I'm not sure we want to see an error! When we first create a production site, we will probably not have any data there until the user creates so data - they'll just see there are currently no articles but they should be able to add some - if there is an error they wont be able to add any!